File: Croce numerica/croce.py
n = 0

# Chiede n fino a che non è un numero dispari positivo
while n%2==0 or n<0:
    n = input("Inserire la dimensione: ")
    n = int(n)

# Crea le righe
for i in range(1, n+1): 
    # Crea le colonne
    for j in range(1, n+1):
        # Scrive solo se è la i-esima o la i-ultima colonna
        if j==i or j==n-i+1:
            print(i, end="")
        else:
            print(" ", end="")
    print()

File: Croce numerica/croce.c
#include <stdio.h>

main(){
    int n;

    // Chiede n fino a che non è un numero dispari positivo
    do{
        printf("Inserire la dimensione: ");
        scanf("%d", &n);
    }while(n%2==0 || n<0);

    // Crea le righe
    for(int i=1; i<=n; i++){
        // Crea le colonne
        for(int j=1; j<=n; j++){
            // Scrive solo se è la i-esima o la i-ultima colonna
            if(j==i || j==n-i+1)
                printf("%d", i);
            else
                printf(" ");
        }

        printf("\n");
    }
}